bokeh/models/widgets/inputs.py

```diff
@@ -89,48 +89,49 @@ def coerce_value(cls, val):
 #-----------------------------------------------------------------------------
 
 class FileInput(Widget):
-    ''' Present a file-chooser dialog to users and return the contents of a
-    selected file.
-
+    ''' Present a file-chooser dialog to users and return the contents of the 
+    selected files 
     '''
-
-    value = String(default="", readonly=True, help="""
-    A base64-encoded string of the contents of the selected file.
-    """)
-
-    mime_type = String(default="", readonly=True, help="""
-    The mime type of the selected file.
-    """)
-
-    filename = String(default="", readonly=True, help="""
-    The filename of the selected file.
-
-    .. note::
-        The full file path is not included since browsers will not provide
-        access to that information for security reasons.
-    """)
+    
+    value = Either(String, List(String), default='', readonly=True, help='''
+    if multiple=False or ommitted, base64-encoded string of the contents of the 
+    selected file,
+    if multiple=True or 'Multiple' a list of base64 encoded strings of the contents 
```

Comment:
comment about `"Multiple"` setting value can be removed (here and below). I might also suggest verbiage with an initial short sentence, then further explanation,e.g. 

```
    value = Either(String, List(String), default='', readonly=True, help='''
    The base64-enconded contents of the file or files that were loaded.

    If `mulitiple` is set to False (default), this value is a single string with the contents
    of the single file that was chosen.

    If `multiple` is True, this value is a list of strings, each contenting the contents of 
    one of the multiple files that were chosen. 
    ''')
``` 

And similar below, what do you think?

Reply:
Applied in the new commit to all three items (value, filename, mime-type).
I changed "contenting" to "containing", although I'm not native speaker, it seemed better.

Reply:
Yes, that was a typo on my part, thanks